connector.go

```diff
@@ -261,10 +260,10 @@ func (c *Connector) startConsumer(overrideOpts ...kafka.ConfigMap) error {
 
 // generateTopicFromProto generates message queue topic names based on the protobuf message.
 // Event names should be prefixed with contract_ or category_ when appropriate.
-func (c *Connector) generateTopicFromProto(msg proto.Message) kafkautils.Topic {
+func (c *Connector) generateTopicFromProto(msgType kafkautils.MsgType, msg proto.Message) kafkautils.Topic {
 	return kafkautils.NewTopic(
 		c.env,
-		c.MsgType,
+		msgType,
```

Comment:
Why not use `c.MsgType`?

Reply:
because it is always set as `fct` when initializing `Connector`. This allows to pass is different type of msg